src/relations/openfga.py

```diff
@@ -121,6 +184,377 @@ def _on_create_authorization_model_action(self, event):
+                    }
+                )
+                return
+            except ApiException:
+                event.set_results({"error": "failed to perform ofga operation"})
+
+
+def _validate_event_params(event, valid_combinations):
+    """Validate event parameters against a list of valid parameter combinations.
+
+    Args:
+        event: The event triggered when the action is performed.
+        valid_combinations: A list of valid parameter combinations, where
+            each combination is represented as a set of parameter keys.
+
+    Returns:
+        bool: True if the provided parameters match any valid combination; False otherwise.
+    """
+    valid = False
+    for valid_combination in valid_combinations:
+        if event.params.keys() == valid_combination:
+            valid = True
+
+    if not valid:
+        valid_combination_strings = []
+        for i, combination in enumerate(valid_combinations):
+            valid_combination_strings.append(f"{i+1}. '{', '.join(combination)}'")
+
+        valid_combinations_message = "\n".join(valid_combination_strings)
+        event.set_results(
+            {
+                "error": f"parameter combination not supported. parameters for this operation must either be:\n{valid_combinations_message}"
+            }
+        )
+
+    return valid
+
+
+def _get_ofga_client(openfga_data):
+    """Create and configure an OpenFGA client.
+
+    Args:
+        openfga_data: Object containing OpenFGA store data.
+
+    Returns:
+        OpenFgaClient: An initialized OpenFgaClient instance configured to interact
+        with the OpenFGA store.
+    """
+    configuration = openfga_sdk.ClientConfiguration(
+        api_scheme=openfga_data["scheme"],
+        api_host=f"{openfga_data['address']}:{openfga_data['port']}",
+        store_id=openfga_data["store_id"],
+        authorization_model_id=openfga_data["auth_model_id"],
+        credentials=Credentials(
+            method="api_token",
+            configuration=CredentialConfiguration(
+                api_token=openfga_data["token"],
+            ),
+        ),
+    )
+
+    return OpenFgaClient(configuration)
+
+
+async def _perform_ofga_api_call(event, openfga_data, body, op_type):
+    """Perform an OpenFGA API call based on the specified operation type.
+
+    Args:
+        event: The event triggered when the action is performed.
+        openfga_data: Object containing OpenFGA store data.
+        body: The request body for the OpenFGA API call, specific to the operation type.
+        op_type: OFGAOperationType operation.
+
+    Returns:
+        Response: The response from the OpenFGA API call, which varies based on the
+        operation type.
+
+    Raises:
+        e: If an error occurs during the OpenFGA API call.
+    """
+    ofga_client = _get_ofga_client(openfga_data)
+
+    try:
+        if op_type == OFGAOperationType.CHECK:
```

Comment:
If you are using Python `3.10<=`  check out `match` statement, this is the perfect place for it.